### ISSUE TITLE
Generator support for generic list/set bindings to array fields

### ISSUE THREAD

**Issue by alexcase52** [jdbi/jdbi#1784]
I have pojo like
```
class SomePojo {
   Long id;
   List<String> field;
}
```
the `field` is mapped into array column in DB. 
I have
```
AppDao {
 @SqlUpdate
   void create(@BindFields SomePojo pojo)
}
```
This is working fine when I use proxy-based SqlObjects. But when I switched to generator, I got errors like:
No argument factory registered for type [java.util.List] for element [field] on [SomePojo@dd568eb] [statement:"INSERT INTO some_pojo(id, field) values (id, :field)", arguments:{positional:{}, named:{}, finder:[{lazy bean field arguments "SomePojo@dd568eb"]}]

I can workaround this by changing pojo to:
```
class SomePojo {
   Long id;
   String[] field;
}
```
but this produces some boilerplate to convert then into `Set<String>` or `List<String>`

jdbi3-generator 3.17.0

**Comment by stevenschlansker** [comment]
Thank you for reporting this, I'm excited someone is actually using the generator!

**Comment by stevenschlansker** [comment]
Hi @alexcase52 , if you have time could you test and see if #1788 fixes your problem?

**Comment by stevenschlansker** [comment]
Also note that you'll need to have `String` registered as an array type, the H2DatabasePlugin and PostgresPlugin do this for you, or you can call `registerArrayType` yourself.

**Comment by stevenschlansker** [comment]
Fixed in 3.18.0
